mmdet/models/dense_heads/reppoints_head.py

```diff
@@ -649,67 +653,66 @@ def loss(self,
         }
         return loss_dict_all
 
-    def get_bboxes(self,
-                   cls_scores,
-                   pts_preds_init,
-                   pts_preds_refine,
-                   img_metas,
-                   cfg=None,
-                   rescale=False,
-                   with_nms=True):
-        assert len(cls_scores) == len(pts_preds_refine)
-        device = cls_scores[0].device
-        bbox_preds_refine = [
-            self.points2bbox(pts_pred_refine)
-            for pts_pred_refine in pts_preds_refine
-        ]
-        num_levels = len(cls_scores)
-        featmap_sizes = [
-            cls_scores[i].size()[-2:] for i in range(len(cls_scores))
-        ]
-        multi_level_points = self.point_generator.grid_priors(
-            featmap_sizes, device)
-
-        result_list = []
-        for img_id in range(len(img_metas)):
-            cls_score_list = [cls_scores[i][img_id] for i in range(num_levels)]
-            bbox_pred_list = [
-                bbox_preds_refine[i][img_id] for i in range(num_levels)
-            ]
-            img_shape = img_metas[img_id]['img_shape']
-            scale_factor = img_metas[img_id]['scale_factor']
-            proposals = self._get_bboxes_single(cls_score_list, bbox_pred_list,
-                                                multi_level_points, img_shape,
-                                                scale_factor, cfg, rescale,
-                                                with_nms)
-            result_list.append(proposals)
-        return result_list
-
+    # Same as base_dense_head/_get_bboxes_single except self._bbox_decode
     def _get_bboxes_single(self,
-                           cls_scores,
-                           bbox_preds,
-                           mlvl_points,
-                           img_shape,
-                           scale_factor,
+                           cls_score_list,
+                           bbox_pred_list,
+                           score_factor_list,
+                           img_meta,
                            cfg,
                            rescale=False,
-                           with_nms=True):
+                           with_nms=True,
+                           **kwargs):
+        """Transform outputs of single image into bbox predictions.
+
+        Args:
+            cls_score_list (list[Tensor]): Box scores for a single image
```

Comment:
Box scores from all scale levels of a single image, each item has shape (num_points * num_classes, H, W).